crates/next-core/js/src/overlay/internal/ReactDevOverlay.tsx

```diff
@@ -50,6 +53,13 @@ function pushErrorFilterDuplicates(
 function reducer(state: OverlayState, ev: Bus.BusEvent): OverlayState {
   switch (ev.type) {
     case Bus.TYPE_BUILD_OK: {
+      if (state.reactError != null) {
+        console.warn(
+          "[Fast Refresh] performing full reload because your application had an unrecoverable error"
+        );
+        window.location.reload();
```

Comment:
I'm guessing this is how Next.js does it, but why not try to re-render the component instead of reloading the window?

Reply:
That's a good question

Reply:
I'll try implementing that

one problem I already see with either implementation right now is the issues from other chunks preventing the `onBuildOk` call

Reply:
hmm I think I might go for the hard reload for now and revisit in the future, running into a bunch of edge cases when I try using it without

the react error boundary is a huge pain